charts/kafka-operator/templates/operator-rbac.yaml

```diff
@@ -8,7 +8,7 @@ imagePullSecrets:
 {{- end }}
 {{- end }}
 metadata:
-  name: {{ include "kafka-operator.fullname" . }}-operator
+  name: {{ .Values.serviceAccount.name }}
```

Comment:
Please move this part under _helpers.tpl and solve the logic there, will result in much cleaner code.
See as an example the Grafana helm chart:
https://github.com/helm/charts/blob/0219c128a8c1be5ea39842e40e69295664001efc/stable/grafana/templates/_helpers.tpl#L34